### PR TITLE
Adding support for snapshot chain name

### PR DESCRIPTION
**Pull request by jesse-l**
Adding support for snapshot chain name in google_compute_snapshot and in google_compute_resource_policy.  This PR is for hashicorp/terraform-provider-google#12429

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added field `chain_name` to resource `google_compute_snapshot` 
```

```release-note:enhancement
compute: added field `chain_name` to resource `google_compute_resource_policy. snapshot_properties` 
```

### CODE REVIEW

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -12944,6 +12944,13 @@ objects:
                   - snapshot_schedule_policy.0.snapshot_properties.0.guest_flush
                 description: |
                   Whether to perform a 'guest aware' snapshot.
+              - !ruby/object:Api::Type::String
+                name: 'chainName'
+                allow_empty_object: true
```
Comment (ScottSuarez) [inline mmv1/products/compute/api.yaml:12949]: `allow_empty_object` field can be removed.. This only [operates on nested fields](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/type.rb#L69-L77). What was your intention in adding this?

File: mmv1/products/compute/api.yaml [hunk 2]
```diff
@@ -14042,6 +14049,16 @@ objects:
...
+      - !ruby/object:Api::Type::String
+        name: 'chainName'
+        allow_empty_object: true
+        description: |
+          Creates the new snapshot in the snapshot chain labeled with the 
+          specified name. The chain name must be 1-63 characters long and 
+          comply with RFC1035. This is an uncommon option only for advanced 
+          service owners who needs to create separate snapshot chains, for 
+          example, for chargeback tracking.  When you describe your snapshot 
+          resource, this field is visible only if it has a non-empty value.
```
Comment (ScottSuarez) [inline mmv1/products/compute/api.yaml:14061]: `allow_empty_object` field can be removed.. This only [operates on nested fields](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/type.rb#L69-L77). What was your intention in adding this?

**Review by ScottSuarez — COMMENTED** [review]
Do you know if this field is updatable?

**Comment by ScottSuarez** [comment]
Looks like the test failed. Let me know if you need any help investigating !

**Comment by jesse-l** [comment]
Yeah it shows it failed the first time test, not sure why that happened, but after I removed the parameters you suggested the second test was success.

**Review by ScottSuarez — APPROVED** [review]
Looks good, thanks for the work!

**Pull request merged**